ledger-transport-tcp/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+use std::{
+    net::{IpAddr, SocketAddr},
+    ops::Deref,
+    time::Duration,
+};
+
+use byteorder::{ByteOrder, NetworkEndian};
+use ledger_transport::{APDUAnswer, APDUCommand, Exchange};
+
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+    sync::Mutex,
+};
+
+mod error;
+pub use error::Error;
+
+/// Ledger speculos (TCP) transport
+pub struct TransportTcp {
+    s: Mutex<TcpStream>,
+    timeout: Duration,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "clap", derive(clap::Parser))]
```

Comment:
I guess this is to make it easier to be used in cli applications by having a subcommand? I think it's unnecessary/overkill, and considering it's such a limited number of options I'd rather have it as arguments to the `new` function...

This also allows us to drop the `humantime` dependency and just have any `Duration` as argument.

As for the defaults, either have options for the arguments and document it, or use a builder-like pattern, which imo is more idiomatic

Reply:
> I guess this is to make it easier to be used in cli applications by having a subcommand?
> considering it's such a limited number of options I'd rather have it as arguments to the new function...

sortof, it makes it easier to embed configuration / provide defaults, and simplifies using the library / helps unify the configuration across different consumers. it's an easy addition and if it's not provided here it has to be implemented by every user of the library.

`clap` is the ecosystem standard for this (now derive has moved from `structopt`), i can switch it from default to opt in?

> As for the defaults, either have options for the arguments and document it, or use a builder-like pattern, which imo is more idiomatic

implementing `Default::default` gives you the same functionality without actually having a builder (which i have now done), and the configuration object is easy to wrap or drop into `clap` (or `serde`, though i haven't needed this -yet-)

Reply:
> it has to be implemented by every user of the library
yes that would be the point, but I find it unlikely that the struct would be used directly anyways, unless it's a very minimal cli... 
I guess removing it from the default set of features is sufficient